src/common/linebuf-reader_test.cc

```diff
@@ -0,0 +1,148 @@
+/* -*- mode: c++; c-basic-offset: 2; indent-tabs-mode: nil; -*-
+ * (c) 2016 Henner Zeller <h.zeller@acm.org>
+ *
+ * This file is part of BeagleG. http://github.com/hzeller/beagleg
+ *
+ * BeagleG is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * BeagleG is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with BeagleG.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "linebuf-reader.h"
+
+#include <string>
+#include <gtest/gtest.h>
+
+// Raw lines that we use as samples.
+constexpr int kSampleLineCount = 3;
+const char *kSampleLines[kSampleLineCount + 1] = {
+  "This is a short line",
+  "This is another, longer line",
+  "",   // Emtpy line
```

Comment:
Just for extra testing: can we have an additional empty line here (with kSampleLineCount = 4) to see if we can walk the state machine in all cases to detect two separate lines here ?
(and probably an additional `EXPECT_EQ(std::string(kSampleLines[3]), reader.ReadLine());` where we read it.